sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -398,7 +398,7 @@ private[spark] object HiveUtils extends Logging {
         case childFirst: ChildFirstURLClassLoader =>
           childFirst.getURLs() ++ allJars(Utils.getSparkClassLoader)
         case urlClassLoader: URLClassLoader =>
-          urlClassLoader.getURLs ++ allJars(urlClassLoader.getParent)
+          allJars(urlClassLoader.getParent) ++ urlClassLoader.getURLs
```

Comment:
Should we guard the change by `spark.driver.userClassPathFirst`?

Reply:
It's already implied by ChildFirstURLClassLoader. Do you want to me check again in URLClassLoader?

Reply:
Ohh okay I got what you mean.